ballista/core/src/execution_plans/shuffle_writer.rs

```diff
@@ -139,7 +154,7 @@ impl ShuffleWriterExec {
         self.shuffle_output_partitioning.as_ref()
     }
 
-    pub fn execute_shuffle_write(
+    fn execute_shuffle_write_internal(
```

Comment:
@tustvold You changed the signature of this method in https://github.com/apache/arrow-ballista/commit/b7bb2cfba13cc04a08c2f687102dd14a8dedc7b6 to change it from `async` to return an `impl Future`, and I am now calling it from an `async` method so just wanted to check that I am not undoing anything here.

Reply:
So there are two changes here

* [`async_trait`] will box the future, so polling the future now goes through a dyn dispatch. This probably doesn't matter given this method is likely performing non-trivial IO
* The returned future borrows self, i.e. can't outlive the borrow of self. This could matter, but given the pervasive use of `Arc` and `async move` it is probably easy enough to work around